src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/ILogger.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// A interface for logging.  It is supported by System.Diagnostics.Tracing.Logger, but it could in theory have
+    /// other implementations.   See Logger for more details on expected use.  
+    /// </summary>
+    public interface ILogger
+    {
+        /// <summary>
+        /// returns true if the given verbosity level is active (will be logged if 'Log' is called).  
+        /// </summary>
+        bool IsEnabled(LogLevel level);
+
+        /// <summary>
+        /// Log a single message.   A message has a name, a Verbosity level and a object that contains any other
+        /// information needed by the message.   Typical this object is created using an anonymous type.  
+        /// </summary>
+
+        void Log(string messageName, LogLevel level, object arguments = null);
```

Comment:
One of the goals stated in the comment below is the ability to use this as a facade on log4net, NLog etc., but none of those frameworks use this style of "name + payload" logging. Compatibility is probably not worth chasing if the experience is second-class/inconsistent with native use of those frameworks.